doc/OnlineDocs/contributed_packages/community.rst

```diff
@@ -0,0 +1,274 @@
+Community Detection for Pyomo models
+====================================
+
+This package separates model components (variables, constraints, and objectives) into different communities
+distinguished by the degree of connectivity between community members.
+
+Description of Package and Main Function `detect_communities`
+-------------------------------------------------------------
+
+The community detection package allows users to obtain a dictionary of the communities in a Pyomo model. The package
+takes in a model, organizes the model components into a graph of nodes and edges, then uses Louvain
+community detection (`Blondel et al, 2008`_) to determine the communities that exist within the model.
+
+.. _Blondel et al, 2008: https://dx.doi.org/10.1088/1742-5468/2008/10/P10008
+
+In graph theory, a community is defined as a subset of nodes that have a greater degree of connectivity within
+themselves than they do with the rest of the nodes in the graph. In the context of Pyomo models, a community
+represents a subproblem within the overall optimization problem. Identifying these subproblems and then solving them
+independently can save computational work as compared with trying to solve the entire model at once. Thus, it
+can be very useful to know the communities that exist in a model.
+
+The manner in which the graph of nodes and edges is constructed from the model directly affects the community
+detection. Thus, this package provides the user with a lot of control over the construction of the graph. The
+parameters of this function that are used for this graph construction and subsequent community detection
+are shown below.
+
+    `detect_communities(model[, type_of_community_map, with_objective, weighted_graph, random_seed])`
+
+The main graph features the user can specify are the type of community map, whether the graph is weighted or
+unweighted, and whether the objective function(s) is included in the graph generation. Below, the significance
+and reasoning behind including each of these options are explained in greater depth.
+
+Type of Community Map
+    In this package's main function (detect_communities), the user can select 'b', 'c', or 'v' as an input for
+    the 'type_of_community_map' argument, and these result in a community map based on a bipartite graph, a constraint
+    node graph, or a variable node graph (respectively).
```

Comment:
Can I suggest instead `nodes_are=['variables'|'constraints'|'variables_and_constraints']`?

I had originally thought `both` instead of `variables_and_constraints`, but it is conceivable to add more classes later on.